### ISSUE TITLE
A custom decorator is not being resolved by the `bundle` command

### ISSUE THREAD

**Issue by MilanLund** [Redocly/redocly-cli#898]
**Describe the bug**
A custom decorator is not being resolved by the `bundle` command.

**To Reproduce**
1. Given this `redocly.yaml` file:
```
apis:
  main: 
    root: openapi/openapi.yaml
    decorators:
      plugin/prepend: warn
plugins:
  - "./plugins/plugin.js"
extends:
  - recommended
decorators:
  plugin/prepend: warn
```
2. And this OpenAPI file: https://github.com/Kontent-ai-Learn/Redocly-API-reference/blob/develop/folder/openapi/openapi.yaml
4. Run this command: `redocly bundle`
5. See that the output does not contain my custom string defined in the decorator in the `Info.description` property.

**The decorator code**
The `./plugins/plugin.js` file:
```
const Prepend = require('./decorators/prepend');
const id = 'plugin';

/** @type {import('@redocly/cli').CustomRulesConfig} */
const decorators = {
  oas3: {
    'prepend': Prepend,
  },
};

module.exports = {
  id,
  decorators,
};
```

The `./plugins/decorators/prepend.js` file:
```
module.exports = Prepend;

/** @type {import('@redocly/cli').OasDecorator} */

function Prepend() {
  return {
    Info: {
      leave(Info) {
        Info.description = `XXXX My custom string XXXX ${Info.description}`
      }
    }
  }
};
```

**Expected behavior**
The decorator gets resolved and therefore "XXXX My custom string XXXX" prepends to the `Info.description` property in the `bundle` command output. The expectation is based on your [documentation](https://redocly.com/docs/cli/commands/bundle/) that mentions that the decorators will get resolved during the `bundle` command execution.

**Logs**
There are no relevant logs.

**OpenAPI definition**
Version: 3.1.0
OpenAPI file:  https://github.com/Kontent-ai-Learn/Redocly-API-reference/blob/develop/folder/openapi/openapi.yaml

**Redocly Versions**
1.0.0-beta.110, 1.0.0-beta.105

**`Node.js` Version**
14.15.1

**Additional context**
What I have found out during my testing:
- In version `1.0.0-beta.110`, the custom decorator gets resolved using the `preview-docs` command.
- In version `1.0.0-beta.110`, [built-in decorators](https://redocly.com/docs/cli/decorators/) get resolved using the `preview-docs` command as well as using the `bundle` command.
- In version `1.0.0-beta.105`, custom as well as built-in decorators do not get resolved using the `bundle` command.

Additional notes:
- The above-stated example of the custom decorator is a simplification of what I am trying to achieve to demonstrate the issue itself.
- We are building and publishing (now evaluating) our API reference using your "API Registry" product. At the time of creating this issue, the build process uses the `1.0.0-beta.105` version. That is a couple of versions behind the current version `1.0.0-beta.110`. Once this issue gets resolved, I would also expect a new version of the package is used for building API Registry in your product.

**Comment by adamaltman** [comment]
Thanks for the detailed issue @MilanLund. 

This has uncovered a bug that I will reported today: #899. 

The issue is the api-specific overrides of anything like `decorators`, `extends`, `rules`, etc results in all root values being ignored for that api (it's not only overriding the `decorator`, for example).

When there are no overrides for a given API, it works as expected:

```
apis:
  repro@issue898: 
    root: openapi.yaml
    # decorators: 
    #   plugin/prepend: warn
plugins:
  - "./plugins/plugin.js"
extends:
  - recommended
decorators:
  plugin/prepend: warn
```

Resulting description (I simplified the OpenAPI further for my reproduction and it starts with a description of "Sample."):

`  description: XXXX My custom string XXXX Sample.`

**Comment by adamaltman** [comment]
As another note, the issue you found was introduced in `1.0.0-beta.109`, but the reason it doesn't resolve for you with `1.0.0-beta.105` is because the configuration file structure changed. It was this then:

```yaml
apis:
  main: 
    root: openapi/openapi.yaml
    lint:
      decorators:
        plugin/prepend: warn
lint:
  plugins:
    - "./plugins/plugin.js"
  extends:
    - recommended
  decorators:
    plugin/prepend: warn
```

**Comment by MilanLund** [comment]
Thank you for the clarification @adamaltman.
I modified my implementation according to your suggestions and things are working now:
- Downgraded from `1.0.0-beta.110` to `1.0.0-beta.105` to be in line with the Redocly API Registry build process.
- Omitted overrides.
- Wrapped the decorator-related lines in the `lint` section in the OpenAPI file.

Now, what seems to be a bit misleading is the way Redocly documentation describes the usage of the decorator. [In step 4](https://redocly.com/docs/cli/guides/replace-server-url/) it says "Register your plugins and `decorators` within the `lint` section.". However, the code sample below does not define the `lint` section. I guess, this issue is not related directly to this repository. I will try to address that with Redocly support.

As things are working for me now, feel free to close this issue if there is nothing more to say or do on your end.

**Comment by adamaltman** [comment]
A few notes. The Redocly API registry builds will use whatever version of Redocly CLI you have pinned in the package.json. So it isn't actually fixed on `1.0.0-beta.105`. 

Yours is using that version because of this line: https://github.com/Kontent-ai-Learn/Redocly-API-reference/blob/develop/folder/package.json#L5

I would recommend the other way around updating to beta.110 and removing overrides for now. Ideally, `1.0.0-beta.111` will be released with a fix for the overrides.

I can fix those docs because the `lint` object was removed (those are also coming from this repo).

**Comment by MilanLund** [comment]
Oh, nice, thanks for more guidance.

**Comment by adamaltman** [comment]
I opened this issue: https://github.com/Redocly/redocly-cli/issues/901 

I have a decent work-in-progress in my environment, but it will be blocked by #899. Once that is fixed, I'll update the docs with a much better and revised guide. This one using environment variables uses an outdated approach (and I wrote it too🤷).

**Comment by adamaltman** [comment]
@MilanLund we released `1.0.0-beta.111` (see the [changelog](https://redocly.com/docs/cli/changelog/#100-beta111-2022-10-10)).  The bug you reported is also fixed in `beta.111`.

To use it, please update your package.json to `1.0.0-beta.111` and run `npm install` to update the corresponding `package-lock.json` file.

I've (significantly) revised the [replace server URLs guide](https://redocly.com/docs/cli/guides/replace-servers-url/) into a follow-along tutorial. I hope that will clear up the best way to use it. It does involve API decorator configuration overrides.

Let us know if you have any problems.
